src/utils/Errors.huff

```diff
@@ -0,0 +1,115 @@
+#define constant EMPTY_ARRAY_POP = 0x31
+#define constant ARRAY_OUT_OF_BOUNDS = 0x32
+#define constant MEMORY_TOO_LARGE = 0x41
+#define constant UNINITIALIZED_FUNCTION_POINTER = 0x51
+
+/*
+
+Solidity Require. Error `string` MUST be no greater than 32 bytes.
+
+MEMORY LAYOUT WHEN THROWN
+| sig    || message offset                                          || message "revert"                                        |
+0x08c379a  000000000000000000000000000000000000000000000000000000020  000000000000000000000000000000000000000000006726576657274
+
+*/
+#define macro REQUIRE() = takes (2) returns (0) {
+    // takes:       // [condition, message]
+    do_not_throw    // [do_not_throw_jumpdest, condition, message]
+    jumpi           // [message]
+    [ERROR_SIG]     // [error_sig, message]
+    0x00            // [mem_ptr, error_sig, message]
+    mstore          // [message]
+    0x20            // [message_offset, message]
+    0x04            // [message_offset_ptr, message_offset, message]
+    mstore          // [message]
+    0x24            // [message_ptr, message]
+    mstore          // []
+    0x44            // [size]
+    0x00            // [offset, size]
+    revert          // []
+    do_not_throw:   // [message]
+        pop         // []
+}
+
+/*
+
+Solidity Panic.
+
+MEMORY LAYOUT WHEN THROWN
+| sig     || panic code                                              |
+0x4e487b71  000000000000000000000000000000000000000000000000000000001
+
+*/
+#define macro PANIC() = takes (1) returns (0) {
+    // takes:       // [panic_code]
+    [PANIC_SIG]     // [panic_sig, panic_code]
+    0x00            // [panic_sig_offset, panic_sig, panic_code]
+    mstore          // [panic_code]
+    0x04            // [panic_code_offset, panic_code]
+    mstore          // []
+    0x24            // [revert_size]
+    0x00            // [revert_offset,, revert_size]
+    revert          // []
+}
+
+/*
+Solidity Assert.
+
+MEMORY LAYOUT WHEN THROWN
+| sig     || assert failed panic code                                |
+0x4e487b71  000000000000000000000000000000000000000000000000000000001
+
+*/
+#define macro ASSERT() = takes (1) returns (0) {
+    // takes:               // [condition]
+    do_not_panic            // [do_not_panic_jumpdest, condition]
+    jumpi                   // []
+    [ASSERT_FALSE]          // [assert_false]
+    PANIC()                 // []
+    do_not_panic:           // []
+}
+
+/* Bubbles up revert data if call failed. Call directly after `call`, `staticcall`, `delegatecall`. */
+#define macro BUBBLE_UP_IF_FAILED() = takes (1) returns (0) {
+    // takes:       // [call_succeeded]
+    call_succeeded  // [call_succeeded_jumpdest, call_succeeded]
+    jumpi           // []
+    returndatasize  // [returndatasize]
+    0x00            // [memory_offset, returndatasize]
+    returndatasize  // [returndatasize, memory_offset, returndatasize]
+    dup2            // [returndata_offset, returndatasize, memory_offset]
```

Comment:
Missing a `returndatasize` at the end of the stack comment here I believe

Reply:
fixed